### DISCUSSION TITLE
GitHub Classroom assignment repos do not have autograder directory (.github)

### DISCUSSION THREAD

**Discussion by mlandis** [community/community#136724] [GitHub Education]
### Hi everyone,

Hi all,

I'm testing my classroom setup for a new semester in 2024. As far as I can tell, repos for accepted assignments do not contain the .github directory used to run GitHub Actions. I saw that GitHub Classroom changed how it runs autograders. I suspect I've missed an important change in the setup protocol, but I can't find what's wrong. Any help is appreciated.

I've made multiple test assignments, with and without templates, and using different types of tests. When I accept the assignment and clone the repo, it does not contain the .github directory, so tests don't run when I commit or when I run tests from the classroom dashboard.

I am following the instructions in the new docs, here: https://docs.github.com/en/education/manage-coursework-with-github-classroom/teach-with-github-classroom/use-autograding. The instructions take steps very similar to those I took when I last ran the class in 2022. 

I read through this post (https://github.com/orgs/community/discussions/106545) where student repos don't reliably copy the autograder files when accepting the assignment. I see that instructors manually create autograding.json and classroom.yml files as a workaround. Is there an tutorial for how to do this?

If any GitHub Classroom support/dev folks read this, here's the classroom assignment:
https://classroom.github.com/classrooms/69019055-practical-bioinformatics-2024/assignments/autograder-test
...and here's a cloned assignment repo: https://github.com/WUSTL-Biol4220/autograder-test-mlandis

Many thanks!
Michael

**Comment by mlandis** [comment]
Small update. I'm attempting the workaround suggested by others, where you provide the template with autograding.js and classroom.yml files. Now I can get the tests to run, however they fail because the Action education/autograding@v1 uses to locate to autograding.js. The workflow sets the path as `/home/runner/work/example/example/.github/classroom/autograding.json` instead of `/home/runner/work/example/.github/classroom/autograding.json`. Screenshot attached.

<img width="817" alt="Screenshot 2024-08-24 at 11 38 10 AM" src="https://github.com/user-attachments/assets/b40cc8ce-32fa-49ad-82e5-5a845ff6df41">

Do I need to add something to my `.github/workflows/classroom.yml` to set the correct path?
```
name: GitHub Classroom Workflow

on: [push]

permissions:
  checks: write
  actions: read
  contents: read

jobs:
  build:
    name: Autograding
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: education/autograding@v1
```

**Comment by ProfWentao** [comment]
I have run into a similar problem. According to ChatGPT, I only need .github/classroom/autograding.json file in the template where the assignment template is generated from. During the assignment creation process, it is supposed to auto generate .github/workflow/classroom-autograding.yml file in the assignment template with "GitHub Preset" option. Not sure if ChatGPT is hallucinating or not since I don't see classroom-autograding.yml file being created on the assignment template. And I don't see any discussion on this file in github documentation. I'll try to create the classroom-autograding.yml file manually on the assignment template and see what happens.
